test/crud-with-options.test.js

```diff
@@ -536,6 +536,54 @@ describe('crud-with-options', function() {
 
   });
 
+  describe('upsertWithWhere', function() {
+
+    beforeEach(seed);
+
+    it('should allow upsertWithWhere(data, cb)', function(done) {
+      User.upsertWithWhere({ name: 'John Smith' }, function() {
+        User.find({ where: { name: 'John Lennon' }}, function(err, data) {
+          should.not.exist(err);
+          data.length.should.equal(1);
+          User.find({ where: { name: 'John Smith' }}, function(err, data) {
+            should.not.exist(err);
+            data.length.should.equal(0);
+            done();
+          });
+        });
+      });
+    });
+
+    it('should allow upsertWithWhere(where, data, cb)', function(done) {
```

Comment:
Nitpick: we prefer to avoid starting test description with `should...`; please see our [style guide documentation](https://github.com/strongloop/loopback-contributor-docs/blob/master/style-guide.md#test-naming)